### DISCUSSION TITLE
Spikes is History

### DISCUSSION THREAD

**Discussion by andyblac** [BottlecapDave/HomeAssistant-OctopusEnergy#636] [General]
@BottlecapDave any idea what causing this ?

<img width="1547" alt="Screenshot 2023-12-15 at 12 47 29" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/cc6ca777-fdfc-4212-abd0-393dbb9aa945">
<img width="1547" alt="Screenshot 2023-12-15 at 12 48 21" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/f26a4817-18fb-4950-a08d-95c577869fc5">
<img width="601" alt="Screenshot 2023-12-15 at 12 50 54" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/e4a25315-98ae-4fa8-926f-395482366975">

**Comment by andyblac** [comment]
<img width="545" alt="Screenshot 2023-12-15 at 12 53 06" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/77936/c7c135f0-abcb-414b-bc0b-dd0019de165c">

**Comment by BottlecapDave** [comment]
The sensor takes the delta between API updates. So my guess is the API is returning invalid data in between updates and then correcting itself. For example the data must have returned for example 5 on update 1, 60 on update 2 and then 5 on update 3. The deltas between these updates would result in the sensor spiking positively and then negatively and then stabilising. I thought I'd caught the issue but apparently not. 

I'm not too sure how to fix this as it's the result of dodgy data coming back from the API and huge increases are valid depending on update frequencies, which by the time it goes negative it's too late to fix the data.

**Comment by andyblac** [comment]
maybe some kind of sanity check ie 50kWh is less than 5mins ie 23:55 0kWh, 00:00 50.26 kWh, is kind of impossible right ?, even 50kWh in between the 30mins update is kind of ridiculous ?, I'm only using about 35-50 KWh a day atm

**Reply by RobinXe** [reply]
I'd suggest that any attempts to shape data, that's being faithfully reported from the API, risk unintended consequences given the wide variety of use-cases, and so should take place on users' own systems. You could achieve filtering of extreme values by utilising Home Assistant's in-built filter platform; they call what you're describing a 'range' filter, but you might be better off with an 'outlier' filter.
